### PR TITLE
Some user management adjustments

### PR DESCRIPTION
**Pull request by deathaxe**
The proposed changes try to align T-SQL and MySQL a bit more with regards to structure and scopes related to user management.

### CODE REVIEW

File: SQL/SQL (basic).sublime-syntax [hunk 1]
```diff
@@ -1087,6 +1144,73 @@ contexts:
...
+        2: constant.other.wildcard.percent.sql
+        3: punctuation.definition.identifier.end.sql
+      pop: 1
+    - match: (`)(%)(`)
+      captures:
+        1: punctuation.definition.identifier.begin.sql
+        2: constant.other.wildcard.percent.sql
+        3: punctuation.definition.identifier.end.sql
+      pop: 1
+    - include: identifier-part
```
Comment (keith-hall) [inline SQL/SQL (basic).sublime-syntax:1212]: I'm curious to understand why this belongs in the `basic` file and not in `MySQL`?

File: SQL/TSQL.sublime-syntax [hunk 2]
```diff
@@ -695,6 +695,47 @@ contexts:
...
+  grant:
+    - meta_prepend: true
+    - match: \b(?i:on)\b
+      scope: keyword.context.resource.tsql
+      push: qualified-identifier
+    - include: with-options
+
+  user-privileges:
+    - meta_prepend: true
+    - match: \b(?i:all(?:\s+permissions)?)\b
```
Comment (keith-hall) [inline SQL/TSQL.sublime-syntax:709]: shouldn't `permissions` be `PRIVILEGES`?
https://learn.microsoft.com/en-us/sql/t-sql/statements/grant-transact-sql?view=sql-server-ver16#syntax
https://learn.microsoft.com/en-us/sql/t-sql/statements/revoke-transact-sql?view=sql-server-ver16#syntax

and if so, I think it is already covered by https://github.com/forkeith/Packages/pull/27/files#diff-532863df530e67398448cfd2730f9f561656b1a802e5ab7605948496b603ebccR980 :)

Comment (deathaxe) [reply SQL/TSQL.sublime-syntax]: Just misread the BNF vs. constants, I guess. Fixed. Thanks.

File: SQL/SQL (basic).sublime-syntax (hunk 1 above)
Comment (deathaxe) [reply SQL/SQL (basic).sublime-syntax]: Moved the `user-identifer` backtto MySQL, keeping simple `expect-user-name` only.

**Review by keith-hall — APPROVED** [review]
Looks good, thanks!

**Pull request merged**